pkg/driver/steinel/hpd/poller.go

```diff
@@ -0,0 +1,49 @@
+package hpd
+
+import (
+	"context"
+	"fmt"
+	"go.uber.org/zap"
+	"time"
+)
+
+type sensor interface {
+	GetUpdate() error
+	GetName() string
+}
+
+type Poller struct {
+	pollInterval time.Duration
+	logger       *zap.Logger
+	sensors      []sensor
+}
+
+func NewPoller(pollInterval time.Duration, logger *zap.Logger, sensors ...sensor) *Poller {
+	if pollInterval <= 0 {
+		pollInterval = time.Second * 60
+	}
+
+	return &Poller{
+		pollInterval: pollInterval,
+		logger:       logger,
+		sensors:      sensors,
+	}
+}
+
+func (p *Poller) startPoll(ctx context.Context) error {
+	ticker := time.NewTicker(p.pollInterval)
+	defer ticker.Stop()
+
+	for {
+		select {
+		case <-ctx.Done():
+			return nil
+		case <-ticker.C:
+			for _, sensor := range p.sensors {
+				if err := sensor.GetUpdate(); err != nil {
```

Comment:
In each sensor's implementation of GetUpdate, they are all performing a GET request to the same URL. The idea of this ticket was to refactor out these identical requests so just one GET request is performed by the driver and then the sensors trait properties are updated based on the response from that